pkg/k8s/service.go

```diff
@@ -91,6 +91,39 @@ func isValidServiceFrontendIP(netIP net.IP) bool {
 	return false
 }
 
+type exposureType struct {
+	isSet   bool
+	svcType slim_corev1.ServiceType
+}
+
+func newExposureType(svc *slim_corev1.Service) (*exposureType, error) {
+	typ, isSet := svc.Annotations[annotation.ServiceTypeExposure]
+	if !isSet {
+		return &exposureType{}, nil
+	}
+
+	svcType := slim_corev1.ServiceType(typ)
+
+	switch svcType {
+	case slim_corev1.ServiceTypeClusterIP:
+	case slim_corev1.ServiceTypeNodePort:
+	case slim_corev1.ServiceTypeLoadBalancer:
+	default:
+		return &exposureType{},
+			fmt.Errorf("not supported type for %q: %s", annotation.ServiceTypeExposure, typ)
+	}
+
+	return &exposureType{isSet, svcType}, nil
+}
+
+func (e *exposureType) expose(t slim_corev1.ServiceType) bool {
+	if !e.isSet {
+		return true
+	}
+
+	return t == e.svcType
+}
```

Comment:
It took me a while to understand all pieces together and see what was the intention of "expose". Can we have some function comments explaining what it does?

Reply:
:+1: added code docs and changed names. hopefully, it's more readable now.